uaclient/util.py

```diff
@@ -227,13 +233,15 @@ def _subp(args: 'Sequence[str]', rcs: 'Optional[List[int]]' = None,
     try:
         proc = subprocess.Popen(
             bytes_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        (out, err) = proc.communicate()
+        (out, err) = proc.communicate(timeout=timeout)
     except OSError:
         try:
             raise ProcessExecutionError(
                 cmd=' '.join(args), exit_code=proc.returncode, stderr=err)
         except UnboundLocalError:
             raise ProcessExecutionError(cmd=' '.join(args))
+    except subprocess.TimeoutExpired:
```

Comment:
I believe the only way this would be raised is if a consumer passed in a timeout. As such, I think we can let the `subprocess` exception bubble out to the callers, because they know definitively that they need to handle it.

I think this would also mean that we would no longer need a generic "timeout" `UserFacingException`; each caller can catch `subprocess.TimeoutExpired` and raise an exception with exactly the message that they want.